.net package refactor, PlaybarFragment, AudioPlayerService

theJenix opened the pull request:
This commit contains a few things:

I moved the network related stuff into the .net package (including the new messages just added)

I added a fragment for the play bar.  I don't believe it's finished, but I'd like to get this into the main line so Elizabeth can start playing with placement/look and feel.

AudioPlayerService and SingleFIleAudioPlayer, to play sounds.

This also contains some hacked bluetooth code, but I'm working on rewriting that whole piece, to fix it once and for all.

theJenix committed:
Moved network related code into com.lastcrusade.fanclub.net package

Added SingleFileAudioPlayer, PlaybarFragment, MediaStoreWrapper, and
TestPlaybarActivity for development and testing of the playbar (the bar at the
bottom of the app).
    Currently, this will respond to a play request by playing the first song on
the phone.
    Next steps are to look at how we queue music, and whether the interface
callback is the right approach

theJenix committed:
Merge branch 'master' of git://github.com/TheLastCrusade/fanClub

Conflicts:
	fanClub/src/com/lastcrusade/fanclub/model/Song.java

Updated PlaybarActivity, MediaStoreWrapper and supporting files to support
Song/SongMetadata split

theJenix committed:
Merge branch 'master' of git://github.com/TheLastCrusade/fanClub

Conflicts:
	README.md
	fanClub/AndroidManifest.xml
	fanClub/res/layout/activity_landing.xml

theJenix committed:
Modified onDeviceFound (in HostActivity) to filter discovered devices to only
ones that support our UUID.  Added method to BluetoothUtils to get the supported
UUIDs of the specified BluetoothDevice

theJenix committed:
Added AudioPlayerService, and some infrastructure around launching/using
services, to support the background processes this app needs.
Modified TestPlaybarActivity to use this service.

theJenix committed:
Merged in from origin (no merge message)

theJenix committed:
Merge branch 'connect_fix'

Conflicts:
	fanClub/src/com/lastcrusade/fanclub/HostActivity.java

added method to attempt to filter out incompatible bluetooth devices

theJenix committed:
Fixed bad merge

theJenix committed:
Merge branch 'master' of git://github.com/TheLastCrusade/fanClub

Conflicts:
	fanClub-test/src/com/lastcrusade/fanclub/net/message/MessengerTest.java
	fanClub/AndroidManifest.xml
	fanClub/res/layout/activity_playlist.xml
	fanClub/res/values/strings.xml
	fanClub/src/com/lastcrusade/fanclub/PlaylistActivity.java
	fanClub/src/com/lastcrusade/fanclub/library/MediaStoreWrapper.java
	fanClub/src/com/lastcrusade/fanclub/model/Song.java

Additional changes:

Migrated PlayMessage, PauseMessage, SkipMessage to .net.message package

dgreenhalgh commented on README.md:
```diff
@@ -1,3 +1,49 @@
 # fanClub
 
 **fanClub** is an open-source Android app that provides a platform to crowdsource music playlist creation among friends.
+
```
You overwrote the README. The GitHub workflow documentation has been migrated to the docs folder

theJenix replied:
Fixed in a new commit.

Jesse Rosalia
GIT MAD, President
Internal Operations at The Agency
Undergraduate Computer Science
Georgia Institute of Technology
jesse.rosalia@gatech.edu
https://github.com/theJenix

On Feb 26, 2013, at 11:21 PM, David Greenhalgh wrote:

> In README.md:
> 
> > @@ -1,3 +1,49 @@
> >  # fanClub
> > 
> >  **fanClub** is an open-source Android app that provides a platform to crowdsource music playlist creation among friends.
> > +
> > You overwrote the README. The GitHub workflow documentation has been migrated to the docs folder
> 
> —
> Reply to this email directly or view it on GitHub.

dgreenhalgh commented on fanClub-test/src/com/lastcrusade/fanclub/net/message/SkipMessageTest.java:
```diff
@@ -1,4 +1,4 @@
-package com.lastcrusade.fanclub.message;
+package com.lastcrusade.fanclub.net.message;
```
I like the idea of moving net specific code to its own package

reidbaker replied:
dito

theJenix committed:
fixed README.md

dgreenhalgh commented on fanClub/project.properties:
```diff
@@ -12,4 +12,3 @@
 
 # Project target.
 target=android-17
-android.library.reference.1=../third_party/SlidingMenuLibrary
```
Should you be removing this reference? I'm not sure, but it could be needed for @ejohnson44's code to run.

theJenix replied:
Hmm…I didn't think I committed that file.  Hang on, I'll fix that.

Jesse Rosalia
GIT MAD, President
Internal Operations at The Agency
Undergraduate Computer Science
Georgia Institute of Technology
jesse.rosalia@gatech.edu
https://github.com/theJenix

On Feb 26, 2013, at 11:28 PM, David Greenhalgh wrote:

> In fanClub/project.properties:
> 
> > @@ -12,4 +12,3 @@
> > 
> >  # Project target.
> >  target=android-17
> > -android.library.reference.1=../third_party/SlidingMenuLibrary
> > Should you be removing this reference? I'm not sure, but it could be needed for @ejohnson44's code to run.
> 
> —
> Reply to this email directly or view it on GitHub.

theJenix committed:
fixed project.properties

bryansills commented on fanClub/res/layout/activity_playlist.xml:
```diff
@@ -0,0 +1,21 @@
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+         />
+<!-- 
```
Do we want to commit commented out code?

theJenix replied:
I left it in as an example of how to use the Playbar fragment...but I can remove it if you want.

dgreenhalgh replied:
:horse:

reidbaker replied:
I say leave it but have the comment say this is how you use a playbar fragment.

theJenix replied:
I just removed that whole file.  There's an activity_test_playbar.xml that has a better example of how it's used, and it's designated test code instead of a remnant from our old layouts.

dgreenhalgh commented on fanClub/src/com/lastcrusade/fanclub/HostActivity.java:
```diff
@@ -96,9 +102,9 @@ private void disconnectAllSockets() {
-        return String.format("%s by %s on their hit album %s", song.getMetadata().getTitle(),
-                song.getMetadata().getArtist(), song.getMetadata().getAlbum());
+    private String formatSong(SongMetadata song) {
+        return String.format("%s by %s on their hit album %s", song.getTitle(),
```
Will this actually be displayed anywhere? We should probably not format the song metadata like this, funny as it is

theJenix replied:
This is a test method that can be removed.  I can remove it now or on a future commit.

dgreenhalgh replied:
"Will we use this to format song metadata in the UI, though?" :horse:

reidbaker replied:
@dgreenhalgh I don't think so. This was used for logging last I checked

theJenix replied:
It was used to test the MediaStoreWrapper, and log the song data that we pulled back.  I'll remove it in a later commit.

bryansills commented on fanClub/res/layout/fragment_playbar.xml:
```diff
@@ -0,0 +1,34 @@
+        android:text="@string/now_playing"
+        android:textAppearance="?android:attr/textAppearanceLarge" />
+
+     <ImageButton
```
We should use the default icons for android. Here is the link to download them.
http://developer.android.com/design/downloads/index.html

reidbaker replied:
second

theJenix replied:
Cool...fixing that this morning.

theJenix replied:
Done...I used the Holo Light buttons from the Action Bar pack.

ejohnson44 replied:
Once this gets pulled in, I will add in the other icons we need if they have not been added yet. I was going to do that with yours anyways, but you doing it works fine

dgreenhalgh commented on fanClub/src/com/lastcrusade/fanclub/PlaylistActivity.java:
```diff
@@ -0,0 +1,41 @@
```
"PlaylistActivity doesn't exist in this modern, post-fragment world" :horse:

reidbaker replied:
but the master has the new fragments and is ready to be pulled in

theJenix replied:
Yea for some reason when I pulled from master, it didn't remove this file.  I'll delete it.

Jesse Rosalia
GIT MAD, President
Internal Operations at The Agency
Undergraduate Computer Science
Georgia Institute of Technology
jesse.rosalia@gatech.edu
https://github.com/theJenix

On Feb 27, 2013, at 12:40 AM, Reid Baker wrote:

> In fanClub/src/com/lastcrusade/fanclub/PlaylistActivity.java:
> 
> > @@ -0,0 +1,41 @@
> > but the master has the new fragments and is ready to be pulled in
> 
> —
> Reply to this email directly or view it on GitHub.

dgreenhalgh commented on fanClub/src/com/lastcrusade/fanclub/TestPlaybarActivity.java:
```diff
@@ -0,0 +1,103 @@
```
"Note: this needs to be worked into CoreActivity in the near future" :horse:

dgreenhalgh commented on fanClub/src/com/lastcrusade/fanclub/audio/SingleFileAudioPlayer.java:
```diff
@@ -0,0 +1,55 @@
+
+    @Override
+    public void skip() {
+        // TODO Auto-generated method stub
```
"What needs to happen here to get this done? Just making sure it wasn't forgotten." :horse:

reidbaker commented:
The png's that are in the drawable folder, shouldn't there be one file in each of the drawable-ldpi, drawable-mdpi, etc... folder.
@ejohnson44 started us off uploading each resolution and I think it is a good idea.

dgreenhalgh commented on fanClub/src/com/lastcrusade/fanclub/model/Song.java:
```diff
@@ -3,9 +3,10 @@
 
     private SongMetadata metadata;
+
+    private String filePath;
```
Does this conflict with any of Sills's playlist code, or am I making this up? Just being cautious

theJenix replied:
It shouldnt...I think the location of the variable declaration moved when I merged our changes together, but otherwise it should be the same.

reidbaker commented on fanClub/res/layout/activity_landing.xml:
```diff
@@ -41,4 +41,15 @@
+        android:layout_above="@+id/txt_greeting"
+        android:layout_centerHorizontal="true"
+        android:layout_marginBottom="86dp"
+        android:text="Test Playbar" />
```
You can ignore this if it is test code but if not lets put the strings in strings.xml

theJenix replied:
It's test code...I left it in to give Lizziejohn an example of how the playbar works, but that whole activity can be junked once she integrates the playbar into the main ui.

theJenix replied:
Rereading my comment from yesterday...I dunno what I was thinking.  This is the landing page with all the buttons, and this button is just to launch the test playbar activity.  The button can be removed once the playbar is integrated into the rest of the app.

dgreenhalgh commented on fanClub/src/com/lastcrusade/fanclub/net/BluetoothNotEnabledException.java:
```diff
@@ -1,4 +1,4 @@
-package com.lastcrusade.fanclub;
+package com.lastcrusade.fanclub.net;
```
Should we have a separate package for exceptions?

theJenix replied:
I prefer to keep exceptions close to the code that throws them, but that's a stylistic thing.

dgreenhalgh replied:
Alright, we'll stick to that convention then

reidbaker commented on fanClub/res/layout/activity_test_playbar.xml:
```diff
@@ -0,0 +1,25 @@
+        android:layout_height="fill_parent"
+        android:layout_alignParentLeft="true"
+        android:layout_alignParentTop="true"
+        android:text="SONGS"
```
hardcoded test

theJenix replied:
This is test code (same as TestPlaylistActivity)...I was playing with layout settings.  I'll add comments to this effect.

reidbaker commented on fanClub/res/layout/activity_test_playbar.xml:
```diff
@@ -0,0 +1,25 @@
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_gravity="bottom"
+        class="com.lastcrusade.fanclub.components.PlaybarFragment" />
```
Do you want to change this to . components.PlaybarFragment

theJenix replied:
I don't believe we can do that in layout files.

reidbaker commented on fanClub/src/com/lastcrusade/fanclub/HostActivity.java:
```diff
@@ -177,37 +183,64 @@ protected void onDeviceFound(BluetoothAdapter adapter, Intent intent) {
-                @Override
-                protected void onConnected(BluetoothSocket socket) {
-                    onConnectedFan(socket);
+        //only connect to devices that can support our service
```
For posterity: This code doesnt work at all but it doesn't break anything
Discovery will be fixed in future **branch** by @theJenix

reidbaker commented on fanClub/src/com/lastcrusade/fanclub/TestPlaybarActivity.java:
```diff
@@ -0,0 +1,103 @@
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        // TODO Auto-generated method stub
```
remove

theJenix replied:
Why do you want this removed?  It was left in as an example to a) use the play bar and b) bind to and call a service.  Will a comment suffice?

Jesse Rosalia
GIT MAD, President
Internal Operations at The Agency
Undergraduate Computer Science
Georgia Institute of Technology
jesse.rosalia@gatech.edu
https://github.com/theJenix

On Feb 27, 2013, at 12:40 AM, Reid Baker wrote:

> In fanClub/src/com/lastcrusade/fanclub/TestPlaybarActivity.java:
> 
> > -        public void onServiceConnected(ComponentName className, IBinder iservice) {
> > -            mService = AudioPlayerService.class.cast(iservice);
> > -            mBound = true;
> > -        }
> >   +
> > -        public void onServiceDisconnected(ComponentName className) {
> > -            mService = null;
> > -            mBound = false;
> > -        }
> >   +
> > -    };
> > -    private ServiceLocator<AudioPlayerService> serviceLocator;
> >   +
> > -    @Override
> > -    protected void onCreate(Bundle savedInstanceState) {
> > -        // TODO Auto-generated method stub
> >   remove
> 
> —
> Reply to this email directly or view it on GitHub.

dgreenhalgh replied:
I think he's referring to the auto-generated method stub

theJenix replied:
Ahh..yea I'll remove that.  Also, I didn't realize (perhaps this was dumb on my part) that my signature would be pasted in the comment here.  I'll stop doing that, lol.

reidbaker commented on fanClub/src/com/lastcrusade/fanclub/TestPlaybarActivity.java:
```diff
@@ -0,0 +1,103 @@
+        super.onCreate(savedInstanceState);
+        super.setContentView(R.layout.activity_test_playbar);
+        
+        //create the service locator for the AudioPlayerService (which will bind to the service, launching if needed).
```
@theJenix we should talk about service locator. I think it would be useful in music library service

theJenix replied:
Absolutely...I wrote it to be a generic service locator.  We can use it for all of our services.

dgreenhalgh commented on third_party/.gitignore:
```diff
@@ -1,3 +0,0 @@
-# Ignore file used to sign apk's
-ABSLibrary/
-SlidingMenuLibrary/
```
These shouldn't be in our source control

theJenix replied:
Hmm...ok, apparently git hosed me on this latest merge.  I'll fix this one too.

reidbaker commented on fanClub/src/com/lastcrusade/fanclub/audio/SingleFileAudioPlayer.java:
```diff
@@ -0,0 +1,55 @@
+ */
+public class SingleFileAudioPlayer implements IPlayer {
+
+    private static final String TAG = SingleFileAudioPlayer.class.getName();
```
Why this and not the string?
If this is the proper way to do things should we start using it other places?

theJenix replied:
This is how I did it on other projects, and it slipped in here.  It's more robust in that it's automatically updated if we change the class name, so I would recommend we use this style of code for the TAG....but I am also cool with changing this here if that's less of a pain.

reidbaker replied:
No lets do it that was from now on.

reidbaker commented on fanClub/src/com/lastcrusade/fanclub/components/PlaybarFragment.java:
```diff
@@ -0,0 +1,115 @@
+    /**
+     * Implement this interface to receive events when the user presses the play/pause button
+     * 
+     * @author thejenix
```
Hey I know this weird but I dont think we should have @author tags in this code. Can you remove them. 
If we need to know who wrote what we can use git blame.

reidbaker replied:
but thanks for javadoc ing

theJenix replied:
I'm curious why you don't like @author tags in the code.  They are designed to indicate the major author or contributor to a class...that may be lost in git blame over lots of commits, no?

reidbaker replied:
Lots of reasons. The major author or contributor is who you should talk to if you have a question with some code or if something breaks. If other people have been changing code then they are who you need to talk to. It moves the system along well. The second reason is a mentality for this project that the code is everyone's. Everyone should understand it an know how it impacts the system.

theJenix replied:
Actually, if other people have been changing the code and it breaks, you probably do want to go back to the primary or major author, which is likely not the last person who touched the file.  To me, the author represents the person who created the file in the first place, or who made a large enough change to become an authority on that.  I don't really want to have to comb thru git blame to find that authority.

I don't believe it impacts the mentality for this project to have authors on files, and on larger projects, it's unreasonable to expect that everyone knows everything about the system.  Keeping some documentation on who should know what is good practice, and this is an easy way to do it.

reidbaker replied:
Fine. I dont like it and I don't think I am alone but it isnt worth arguing about. I won't have any @author tags in my code unless the team decides we want it.

theJenix replied:
Let's try and get consensus about this at today's meeting.  For the mean time I'll leave them in, and can change them on the next commit based on what we decide.

reidbaker commented on fanClub/src/com/lastcrusade/fanclub/components/PlaybarFragment.java:
```diff
@@ -0,0 +1,115 @@
+                    PlaybarFragment.this.play();
+                }
+                
+//                transitionTo(HostActivity.class);
```
remove or comment why it is commented out

bryansills replied:
Seconded on Reid's comment. Still pulling it in.

reidbaker commented on fanClub/src/com/lastcrusade/fanclub/components/PlaybarFragment.java:
```diff
@@ -0,0 +1,115 @@
+        });
+
+        ((ImageButton) view.findViewById(R.id.btn_skip))
+        .setOnClickListener(new OnClickListener() {
```
indent

reidbaker commented on fanClub/src/com/lastcrusade/fanclub/service/AudioPlayerService.java:
```diff
@@ -0,0 +1,58 @@
+ * An implementation of Service to provide background music playing functionality.  This uses ILocalBinder
+ * to implement a local binder that enables binding activities to get access to the service.
+ * 
+ * @author Jesse Rosalia
```
remove

reidbaker commented on fanClub/src/com/lastcrusade/fanclub/service/PlayService.java:
```diff
@@ -0,0 +1,21 @@
+/**
+ * This service is responsible for holding the play queue, and feeding songs to the Audio player service.
+ * 
+ * @author Jesse Rosalia
```
remove

reidbaker commented on fanClub/src/com/lastcrusade/fanclub/service/ServiceLocator.java:
```diff
@@ -0,0 +1,55 @@
+ * happens in the message loop, so you must return from the on* method for the service to actually launch.
+ * After the service is bound, use getService to get the service object.
+ * 
+ * @author Jesse Rosalia
```
remove

reidbaker commented on fanClub/src/com/lastcrusade/fanclub/service/ServiceNotBoundException.java:
```diff
@@ -0,0 +1,19 @@
+/**
+ * Attempting to get a service that has not been bound (@see ServiceLocator)
+ * 
+ * @author Jesse Rosalia
```
remove

theJenix replied:
See my disagreement above.  I don't think git blame completely replaces this.

reidbaker commented on third_party/.gitignore:
```diff
@@ -1,3 +0,0 @@
-# Ignore file used to sign apk's
```
I dont think we should have other people code commited into our project. Lets add this back.

theJenix committed:
Added .gitignore back in

theJenix commented on fanClub/src/com/lastcrusade/fanclub/util/BluetoothUtils.java:
```diff
@@ -65,4 +72,34 @@ public static void enableDiscovery(Context context) {
+     * @param device
+     * @return A list of UUIDs supported on the remote system.  Note that this may be empty (but should never be null).
+     */
+    public static List<UUID> getUuidsForDevice(BluetoothDevice device) {
```
For posterity: a remnant of the old new discovery process.  This is all getting replaced within the next couple of days.

theJenix committed:
Replaced playbar icons with Google's Holo Light icons
Fixed bug in SingleFileAudioPlayer pause function, and implemented skip
Removed PlaylistActivity and xml file
Removed unused imports from AcceptThread and ConnectThread

theJenix committed:
Merge branch 'master' of git://github.com/TheLastCrusade/fanClub

Conflicts:
	fanClub/AndroidManifest.xml

theJenix committed:
Added comment for test code/xml

theJenix committed:
Fixed compile error in TestPlaybarActivity due to changes to MediaStoreWrapper
Fixed indenting issues in LandingActivity and PlaybarFragment

theJenix commented:
I believe I've addressed all of the issues in the pull request, except the @author issue.  I can correct that issue in a later commit if we want to pull this in.
